crates/ef-testing/src/evm_sequencer/evm_state/v0.rs

```diff
@@ -33,20 +38,40 @@ impl Evm for KakarotSequencer {
         );
 
         // Set the base fee.
-        let low_fee = base_fee & U256::from(u128::MAX);
-        let low_fee: u128 = low_fee.try_into().unwrap(); // safe unwrap <= U128::MAX.
-        let high_fee = base_fee >> U256::from(128);
-        let high_fee: u128 = high_fee.try_into().unwrap(); // safe unwrap <= U128::MAX.
-
-        let base_address = get_storage_var_address("base_fee", &[]);
+        let [low_fee, high_fee] = split_u256(base_fee);
+        let basefee_address = get_storage_var_address("base_fee", &[]);
         self.state_mut()
-            .set_storage_at(kakarot_address, base_address, StarkFelt::from(low_fee));
+            .set_storage_at(kakarot_address, basefee_address, StarkFelt::from(low_fee));
         self.state_mut().set_storage_at(
             kakarot_address,
-            next_storage_key(&base_address)?,
+            next_storage_key(&basefee_address)?,
             StarkFelt::from(high_fee),
         );
 
+        // Set the previous randao.
+        let [low_prev_randao, high_prev_randao] = split_u256(prev_randao);
+        let prev_randao_address = get_storage_var_address("prev_randao", &[]);
+        self.state_mut().set_storage_at(
+            kakarot_address,
+            prev_randao_address,
+            StarkFelt::from(low_prev_randao),
+        );
+        self.state_mut().set_storage_at(
+            kakarot_address,
+            next_storage_key(&prev_randao_address)?,
+            StarkFelt::from(high_prev_randao),
+        );
+
+        // Set the block gas limit, considering it fits in a felt.
+        let block_gaslimit_as_hexstring = block_gaslimit.to_be_bytes();
+        let block_gaslimit = StarkFelt::new(block_gaslimit_as_hexstring).unwrap();
```

Comment:
```suggestion
        let [block_gas_limit, _] = split_u256(block_gas_limit);
        let block_gaslimit = StarkFelt::from(block_gas_limit);
```

Reply:
I don't want it as u256, and I want to keep eventual bytes above the 128th one

Reply:
even though in practice it's very unlikely to be > 128 bits I think felt from bytes is a correct approach here

Reply:
either it's u256 and then we need to store it in two storage slots, or it's u128 and then we split it. There isn't really any different values in the EVM

Reply:
fyi in the `SealedHeader` type from Reth, they use `u64`.

Reply:
ok 👍